operations/task_output.go

```diff
@@ -0,0 +1,234 @@
+package operations
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/utility"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+const (
+	taskIDFlagName    = "task_id"
+	executionFlagName = "execution"
+
+	logStartFlagName         = "start"
+	logEndFlagName           = "end"
+	logLineLimitFlagName     = "line_limit"
+	logTailLimitFlagName     = "tail_limit"
+	logPrintTimeFlagName     = "print_time"
+	logPrintPriorityFlagName = "print_priority"
+	logPaginateFlagName      = "paginate"
+	logOutputFileFlagName    = "out"
+)
+
+func Task() cli.Command {
+	flags := []cli.Flag{
+		cli.StringFlag{
+			Name:     taskIDFlagName,
+			Usage:    "ID of the task.",
+			Required: true,
+		},
+		cli.IntFlag{
+			Name:  executionFlagName,
+			Usage: "The 0-based number corresponding to the execution of the task ID.",
+		},
+	}
+
+	return cli.Command{
+		Name:  "task",
+		Usage: "operations for Evergreen tasks",
+		Subcommands: []cli.Command{
+			taskBuild(flags),
+		},
+	}
+}
+
+func taskBuild(flags []cli.Flag) cli.Command {
+	return cli.Command{
+		Name:  "build",
+		Usage: "operations for downloading task (build) output",
+		Subcommands: []cli.Command{
+			taskLogs(flags),
+			testLogs(flags),
+		},
+	}
+}
+
+func taskLogs(flags []cli.Flag) cli.Command {
+	const logTypeFlagName = "type"
+
+	return cli.Command{
+		Name:  "TaskLogs",
+		Usage: "download task logs",
+		Flags: mergeFlagSlices(
+			flags,
+			[]cli.Flag{
+				cli.StringFlag{
+					Name:  logTypeFlagName,
+					Usage: "Task log type. Must be one of: \"agent_log\", \"system_log\", \"task_log\", \"all_logs\".",
+					Value: "all_logs",
+				},
+			},
+			taskOutputLogFlags(),
+		),
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			conf, err := NewClientSettings(c.Parent().String(confFlagName))
+			if err != nil {
+				return errors.Wrap(err, "loading configuration")
+			}
+
+			restClient, err := conf.setupRestCommunicator(ctx, false)
+			if err != nil {
+				return errors.Wrap(err, "setting up REST communicator")
+			}
+			defer restClient.Close()
+
+			r, err := restClient.GetTaskLogs(ctx, client.GetTaskLogsOptions{
+				TaskID:        c.String(taskIDFlagName),
+				Execution:     utility.ToIntPtr(c.Parent().Int(executionFlagName)),
```

Comment:
if the execution isn't specified will this pass 0 as the execution?

Reply:
Yep! It defaults to 0, so we always pass something.

Reply:
For the HTTP route we default to the latest execution, but it was too annoying to do here since there aren't optional flags.

Reply:
Ah, I understand better now.
Could it work to check if the flag is set with [IsSet](https://github.com/urfave/cli/blob/1c29f69445b36de39dd2565c80d5328e6553354f/context.go#L55)?

Reply:
Great idea! The only problem is that I can't figure out how to suppress the default message in the parameter description, so it would be confusing to users.

Reply:
Ah, I understand what you're saying.
Maybe it could work as an IntSlice?